clang/docs/checkedc/Setup-and-Build.md

```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
+                                                 // build a package for installation on other
+                                                 // machines.
+   -DLLVM_TARGETS_TO_BUILD="X86"                 // By default, CMake will produce a build
+                                                 // system that builds code generators for all
+                                                 // LLVM-supported architectures. Specify
+                                                 // architecture to decrease build/link times.
+
+   -DLLVM_LIT_ARGS=-v                            // Arguments to pass to the test framework
+   <WORK_DIR>/src/llvm
+   ```
+
+7. After executing the `cmake` command as above, build the compiler as follows:
+
+   ```
+   ninja            // This command will build the compiler and all other supporting tools.
+   OR
+   ninja clang      // This command will build only the compiler.
+
+   ninja clean      // This command cleans the build directory.
+   ```
+
+   ​
+
+### Instruction for Windows (Command shell)
+
+1. Choose any directory as your working directory. We will refer to to this
+directory as \<WORK_DIR\>.
+
+   ```
+   cd <WORK_DIR>
+   ```
+
+2. Clone the `checkedc-clang` repository: If you already have
+`core.autocrlf=false` set for your global Git configuration, you can follow the
+Unix/Linux directions. Otherwise, follow these directions:
+
+   ```
+   git clone -c core.autocrlf=false https://github.com/Microsoft/checkedc-clang src
+   ```
+
+3. The Checked C language tests live in a folder within `llvm\project`. Change
+to the `src\llvm\projects\checkedc-wrapper` directory and clone the Checked C
+repo:
+
+   ```
+   git clone https://github.com/Microsoft/checkedc
+   ```
+
+4. LLVM and Clang use CMake, which is a meta-build system generator. It
+generates build systems for a specific platform. Create a build directory that
+is a sibling of your llvm source tree, like \<WORK_DIR\>\build. Be sure to
+exclude the build directory from anti-virus scanning. On Windows 10, go to
+`Settings->Update & Security->Windows Defender->Add an exclusion`.
+
+      ```
+      cd <WORK_DIR>\build
+      ```
+
+5. In a Windows command shell, execute the commands given below. NOTE: In the
+last command, supply the argument `x64` to build the X64 version of `clang`, or
+the argument `x86` to build the X86 version of clang, or the argument `x86_64`
+to build the X86 version of clang that executes on X64 Windows.
+
+      ```
+      set TOP=<WORK_DIR>
+      cd %TOP%\build
+      set PATH="C:\GnuWin32\bin";%PATH%
+      @call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+      ```
+
+6. Execute the following `cmake` command in the same command shell created above
+in the `<WORK_DIR>\build` directory:
+
+      ```
+      cmake -G Ninja -DLLVM_ENABLE_PROJECTS=clang   // Required to enable Clang build
```

Comment:
We can describe these CMake options once to avoid repetition and maintenance.

Reply:
In my experience, it is extremely helpful for first-time users to be provided command lines that they can virtually copy-paste in their setup. First-timers are usually overwhelmed by the number of unknown/unclear things and need instructions that do not have numerous redirections. So, I think that this repetition is useful and worth the extra maintenance effort.